pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -168,18 +169,33 @@ func (r *ReconcileClusterDisruption) reconcilePDBsForOSDs(
 		}
 		return fmt.Errorf("could not check cluster health: %+v", err)
 	}
+	if len(pdbStateMap.Data) < 1 {
```

Comment:
To clarify this condition, should it be `if pdbStateMap.Data == nil`? Or why would we create a new map if the length is 0?

Reply:
It's equivalent since it's parsed from yaml. Can change.

Reply:
should we also have `pdbStateMap!=nil` ?

Reply:
@rohantmp If it's parsed from yaml, that means it will never be nil? Then there is no need to initialize it here?

Reply:
It might be nil because the field is omitempty.

Reply:
Switched to a nil check instead of a len check anyway

Reply:
> should we also have pdbStateMap!=nil check
No need, this is only to initialize the map if it's nil. The code after assumes it's not nil